src/main/java/com/telesign/RestClient.java

```diff
@@ -20,330 +20,358 @@
+			writeTimeout = 10;
+		}
+
+		OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder()
+				.connectTimeout(connectTimeout, TimeUnit.SECONDS).readTimeout(readTimeout, TimeUnit.SECONDS)
+				.writeTimeout(writeTimeout, TimeUnit.SECONDS);
+
+		if (proxy != null) {
+			okHttpClientBuilder.proxy(proxy);
+
+			if (proxyUsername != null && proxyPassword != null) {
+
+				Authenticator proxyAuthenticator = new Authenticator() {
+					public Request authenticate(Route route, Response response) throws IOException {
+						String credential = Credentials.basic(proxyUsername, proxyPassword);
+						return response.request().newBuilder().header("Proxy-Authorization", credential).build();
+					}
+				};
+
+				okHttpClientBuilder.proxyAuthenticator(proxyAuthenticator);
+			}
+		}
+
+		this.client = okHttpClientBuilder.build();
+	}
+
+	/**
+	 * A simple HTTP Response object to abstract the underlying OkHttp library
+	 * response.
+	 */
+	public static class TelesignResponse {
+
+		public int statusCode;
+		public Map<String, List<String>> headers;
+		public String body;
+		public boolean ok;
+		public JsonObject json;
+
+		public TelesignResponse(Response okHttpResponse) {
+
+			this.statusCode = okHttpResponse.code();
+			this.headers = okHttpResponse.headers().toMultimap();
+			this.ok = okHttpResponse.isSuccessful();
+
+			try {
+				this.body = okHttpResponse.body().string();
+
+				try {
+					this.json = new JsonParser().parse(body).getAsJsonObject();
+				} catch (JsonParseException | IllegalStateException e) {
+					this.json = new JsonObject();
+				}
+
+			} catch (IOException e) {
+				this.body = "";
+				this.json = new JsonObject();
+			}
+		}
+	}
+
+	/**
+	 * Generates the TeleSign REST API headers used to authenticate requests.
+	 * <p>
+	 * Creates the canonicalized stringToSign and generates the HMAC signature. This
+	 * is used to authenticate requests against the TeleSign REST API.
+	 * <p>
+	 * See https://developer.telesign.com/docs/authentication for detailed API
+	 * documentation.
+	 *
+	 * @param customerId
+	 *            Your account customer_id.
+	 * @param apiKey
+	 *            Your account api_key.
+	 * @param methodName
+	 *            The HTTP method name of the request as a upper case string, should
+	 *            be one of 'POST', 'GET', 'PUT' or 'DELETE'.
+	 * @param resource
+	 *            The partial resource URI to perform the request against.
+	 * @param urlEncodedFields
+	 *            URL encoded HTTP body to perform the HTTP request with.
+	 * @param dateRfc2616
+	 *            (optional) The date and time of the request formatted in rfc 2616.
+	 * @param nonce
+	 *            (optional) A unique cryptographic nonce for the request.
+	 * @param userAgent
+	 *            (optional) User Agent associated with the request.
+	 * @return Map of HTTP headers to be applied to the request.
+	 */
+	public Map<String, String> generateTelesignHeaders(String methodName, String resource, String requestParams,
```

Comment:
In each of the SDKs the generateTelesignHeaders (or equivalent) function was designed to be static and not require any 3rd party libraries/packages. The reason for this was to allow customers to extract the function implementation as is and implement it in their application without having to use the SDK and its dependencies.